### PR TITLE
Analytics data uploader 

### PR DESCRIPTION
**Pull request by sadhansood**
## Description 

Add an uploader for analytics base data. This writes data to local storage before uploading to remote. Allows us to configure different file formats (but csv to begin with which already exists).

### CODE REVIEW

File: crates/sui-analytics-indexer/src/lib.rs [hunk 1]
```diff
@@ -26,15 +59,269 @@ pub struct AnalyticsIndexerConfig {
...
+impl Manifest {
+    pub fn new(epoch: u64, next_checkpoint_seq_num: u64) -> Self {
+        Manifest::V1(ManifestV1 {
+            version: 1,
+            next_checkpoint_seq_num,
+            file_metadata: vec![],
+            epoch,
+        })
+    }
+    pub fn files(&self) -> Vec<FileMetadata> {
```
Comment (dariorussi) [inline crates/sui-analytics-indexer/src/lib.rs:251]: same all over here for the newlines, but really up to you

**Review by dariorussi — APPROVED** [review]
really like this! 
thanks and let's move forward!!!

File: crates/sui-analytics-indexer/src/lib.rs [hunk 2]
```diff
@@ -26,15 +59,269 @@ pub struct AnalyticsIndexerConfig {
...
+        match self {
+            FileType::Checkpoint => Path::from(CHECKPOINT_DIR_PREFIX),
+            FileType::Transaction => Path::from(TRANSACTION_DIR_PREFIX),
+            FileType::TransactionObjects => Path::from(TRANSACTION_OBJECT_DIR_PREFIX),
+            FileType::Object => Path::from(OBJECT_DIR_PREFIX),
+            FileType::Event => Path::from(EVENT_DIR_PREFIX),
+            FileType::MoveCall => Path::from(MOVE_CALL_PREFIX),
+        }
+    }
+    pub fn file_path(
```
[the remark this answers is not in the document]
Comment (sadhansood) [reply crates/sui-analytics-indexer/src/lib.rs:147]: done

File: crates/sui-analytics-indexer/src/lib.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-analytics-indexer/src/lib.rs:255]: added everywhere

**Review by longbowlu — APPROVED** [review]
curious is there a way to directly write/stream data to snowflake? i.e. we can configure snowflake a remote store here.

**Comment by sadhansood** [comment]
> curious is there a way to directly write/stream data to snowflake? i.e. we can configure snowflake a remote store here.

There is actually https://docs.snowflake.com/en/user-guide/data-load-snowpipe-streaming-kafka It seems like the most preferred way would be to write to a Kafka topic and snowpipe can stream it from there. We could definitely also give Kafka a try in future and it'd be more realtime for sure (at some extra cost). Right now with the default settings we will see a delay of ~10 mins in checkpoint upload (which can be reduced as well).

**Pull request merged**